### PR TITLE
fix AWS tagging to add missing tags only

### PR DESCRIPTION
**Pull request by redbaron**
It seems that intention of original code was to build map of missing
tags and call AWS API to add just them, but due to typo full
set of tags was always (re)added

```release-note
NONE
```

### CODE REVIEW

**Comment by justinsb** [comment]
@k8s-bot ok to test

Error introduced (by me) in 0b5ae5391ef9eeba337f1dfa62b6f3125e28c86c

/assign

File: pkg/cloudprovider/providers/aws/tags.go [hunk 1]
```diff
@@ -174,7 +174,7 @@ func (c *awsTagging) readRepairClusterTags(client EC2, resourceID string, lifecy
 		}
 	}
 
-	if err := c.createTags(client, resourceID, lifecycle, additionalTags); err != nil {
+	if err := c.createTags(client, resourceID, lifecycle, addTags); err != nil {
```
Comment (justinsb) [inline pkg/cloudprovider/providers/aws/tags.go]: Nice spot

File: pkg/cloudprovider/providers/aws/tags.go [hunk 2]
```diff
@@ -184,9 +184,7 @@ func (c *awsTagging) readRepairClusterTags(client EC2, resourceID string, lifecy
 // createTags calls EC2 CreateTags, but adds retry-on-failure logic
 // We retry mainly because if we create an object, we cannot tag it until it is "fully created" (eventual consistency)
 // The error code varies though (depending on what we are tagging), so we simply retry on all errors
-func (t *awsTagging) createTags(client EC2, resourceID string, lifecycle ResourceLifecycle, additionalTags map[string]string) error {
-	tags := t.buildTags(lifecycle, additionalTags)
```
Comment (justinsb) [inline pkg/cloudprovider/providers/aws/tags.go]: createTags is called in two (?) places in aws.go, and I think we do need buildTags in those cases.

**Review by justinsb — CHANGES_REQUESTED** [review]
Thanks for the fix!  But I don't think we can drop buildTags from createTags, when it is not called from readRepairClusterTags

**Comment by redbaron** [comment]
@justinsb , you are right. I restored call to `buildTags` in `createTags` so other users are not affected. It kinda defeated the purpose of microoptimization in `readRepairClustertags`  as all the tags will be created,so I add len(addTask) check to not call `createTags` if there is nothing to do

**Comment by redbaron** [comment]
BTW, if `additionalTags` is meant to be coming lets say from user at some point, then logic in `readRepairClusterTags` is broken as it wont let additional tags to change values. 

It is not a bug currently as `readRepairClusterTags` is called only once and that call is with `additionalTags=nil`

**Comment by wojtek-t** [comment]
@k8s-bot pull-kubernetes-unit test this

**Comment by justinsb** [comment]
@k8s-bot pull-kubernetes-e2e-kops-aws test this

**Comment by jsafrane** [comment]
@k8s-bot pull-kubernetes-e2e-kops-aws test this

**Pull request merged**